mint-auditor/src/gnosis/sync.rs

```diff
@@ -0,0 +1,549 @@
+        };
+
+        // Perform a gnosis sync.
+        let sync = GnosisSync::new(audited_safe_config, mint_auditor_db, logger).unwrap();
+        sync.process_transactions(raw_transactions);
+
+        // Validate that we are seeing the expected deposits.
+        let deposits = gnosis_safe_deposits::table
+            .order_by(gnosis_safe_deposits::eth_block_number)
+            .load::<GnosisSafeDeposit>(&conn)
+            .unwrap();
+        let expected_deposits = vec![
+            GnosisSafeDeposit {
+                id: Some(2),
+                eth_tx_hash: "0xa202a4c37f0670557ceeb33f796fba0c187f699f5dd4d8add0eba1c3154b2fa7"
+                    .to_string(),
+                eth_block_number: 10824613,
+                safe_address: SAFE_ADDRESS.to_string(),
+                token_address: ETH_TOKEN_CONTRACT_ADDRESS.to_string(),
+                amount: 1000000,
+            },
+            GnosisSafeDeposit {
+                id: Some(1),
+                eth_tx_hash: "0x4f3124c61c48aa7c7892f8fe426e0c0d8afae100fc0a9aa8e290e530a7632849"
+                    .to_string(),
+                eth_block_number: 10824662,
+                safe_address: SAFE_ADDRESS.to_string(),
+                token_address: ETH_TOKEN_CONTRACT_ADDRESS.to_string(),
+                amount: 10000000,
+            },
+        ];
+        assert_eq!(deposits, expected_deposits);
+
+        // Validate that we are seeing the expected withdrawals.
+        let withdrawals = gnosis_safe_withdrawals::table
+            .order_by(gnosis_safe_withdrawals::eth_block_number)
+            .load::<GnosisSafeWithdrawal>(&conn)
+            .unwrap();
+        let expected_withdrawals = vec![
+            GnosisSafeWithdrawal {
+                id: Some(2),
+                eth_tx_hash: "0x323b145662d2a64de0a55977089b7a89ed6003e341d5a68266a200dde83639d4"
+                    .to_string(),
+                eth_block_number: 10824635,
+                safe_address: SAFE_ADDRESS.to_string(),
+                token_address: ETH_TOKEN_CONTRACT_ADDRESS.to_string(),
+                amount: 500000,
+                mc_tx_out_public_key_hex:
+                    "0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20".to_string(),
+            },
+            GnosisSafeWithdrawal {
+                id: Some(1),
+                eth_tx_hash: "0x2f55d7b7620876c1dfc25419937a7fd2538489c1dd3adf6b438396a958d88e28"
+                    .to_string(),
+                eth_block_number: 10824678,
+                safe_address: SAFE_ADDRESS.to_string(),
+                token_address: ETH_TOKEN_CONTRACT_ADDRESS.to_string(),
+                amount: 2000000,
+                mc_tx_out_public_key_hex:
+                    "0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20".to_string(),
+            },
+        ];
+        assert_eq!(withdrawals, expected_withdrawals);
+    }
+
+    #[test_with_logger]
+    fn process_transactions_ignores_unknown(logger: Logger) {
```

Comment:
At the moment I can't think of any, but are there any other edge cases we should be testing here?